src/sensai/hyperopt.py

```diff
@@ -236,19 +219,13 @@ def collectResult(values):
             paramsMetricsCollection.addValues(values)
             _log.info(f"Updated grid search result:\n{paramsMetricsCollection.getDataFrame().to_string()}")
 
-        if self.numProcesses == 1:
-            for parameterOptions in self.parameterOptionsList:
-                for i, paramsDict in enumerate(iterParamCombinations(parameterOptions)):
-                    collectResult(self._evalParams(self.modelFactory, evaluatorOrValidator, self.parameterCombinationSkipDecider, **paramsDict))
-        else:
-            executor = ProcessPoolExecutor(max_workers=self.numProcesses)
-            futures = []
-            for parameterOptions in self.parameterOptionsList:
-                for i, paramsDict in enumerate(iterParamCombinations(parameterOptions)):
-                    futures.append(executor.submit(self._evalParams, self.modelFactory, evaluatorOrValidator, self.parameterCombinationSkipDecider,
-                        **paramsDict))
-            for i, future in enumerate(futures):
-                collectResult(future.result())
+        executor = ProcessPoolExecutor(max_workers=self.numProcesses)
```

Comment:
Why would you remove the option to not use multi-processing?

Reply:
You stated: "It increases the code complexity and I can't see any benefit from it"
There is an obvious benefit to not using multi-processing: No need to send anything across (via pickling) to another process. It is much better to use the main process to do everything.

Reply:
I guess I underestimated the overhead of multiprocessing. You are right, it's not worth it just for getting rid of some lines of code. Since the only remaining changes are cosmetic, I'm closing this PR